src/libs/pluginprocess/pluginprocess.c

```diff
@@ -0,0 +1,379 @@
+	Key * resultPipeKey = keyNew ("/pluginprocess/resultPipe", KEY_VALUE, pp->resultPipe, KEY_END);
+	ELEKTRA_LOG ("P: We send %zd keys through the pipe now to issue command %d it", ksGetSize (keySet), command);
+	elektraInvoke2Args (pp->dump, "set", keySet, commandPipeKey);
+	ELEKTRA_LOG_DEBUG ("P: Waiting for the result now");
+	elektraInvoke2Args (pp->dump, "get", keySet, resultPipeKey);
+	ELEKTRA_LOG ("P: We received %zd keys in return", ksGetSize (keySet));
+
+	// Bring everything back in order by removing our process-related keys
+	Key * originalDeserializedKey = ksLookupByName (keySet, keyName (key), KDB_O_POP);
+	Key * resultKey = ksLookupByName (keySet, "/pluginprocess/result", KDB_O_POP);
+	char * endPtr;
+	int prevErrno = errno;
+	errno = 0;
+	int result = (int)strtol (keyString (resultKey), &endPtr, 10);
+	// Copy everything back into the actual keysets
+	keyCopy (key, originalDeserializedKey);
+	ksCopy (originalKeySet, keySet);
+
+	// Command finished, cleanup now
+	keyDel (commandPipeKey);
+	keyDel (resultPipeKey);
+	keyDel (originalDeserializedKey);
+	ksDel (keySet);
+
+	if (*endPtr != '\0' || errno == ERANGE)
+	{
+		ELEKTRA_ADD_WARNINGF (191, originalKey, "Received invalid return code or no KeySet: %s", keyString (resultKey));
+		keyDel (resultKey);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+	keyDel (resultKey);
+	errno = prevErrno;
+
+	return result;
+}
+
+/** Check if a given plugin process is the parent or the child process
+ *
+ * @param pp the data structure containing the plugin's process information
+ * @retval 0 if it's the child process
+ * @retval the child process' pid otherwise
+ * @ingroup processplugin
+ **/
+int elektraPluginProcessIsParent (const ElektraPluginProcess * pp)
+{
+	return pp->pid;
+}
+
+static int elektraPluginProcessFork (ElektraPluginProcess * pp, Key * errorKey)
+{
+	// Prepare a named pipe so the dump plugin can serialize to it
+	// Create one pipe per plugin instance / handle
+	int mkfifoRet;
+	if ((mkfifoRet = mkfifo (pp->commandPipe, S_IRUSR | S_IWUSR)) == -1)
+	{
+		ELEKTRA_SET_ERRORF (190, errorKey, "Failed to initialize the command pipe, mkfifo () returned %d", mkfifoRet);
+		return -1;
+	}
+	if ((mkfifoRet = mkfifo (pp->resultPipe, S_IRUSR | S_IWUSR)) == -1)
+	{
+		ELEKTRA_SET_ERRORF (190, errorKey, "Failed to initialize the result pipe, mkfifo () returned %d", mkfifoRet);
+		return -1;
+	}
+
+	pid_t pid = fork ();
+	if (pid < 0)
+	{
+		ELEKTRA_SET_ERRORF (190, errorKey, "Failed to fork the plugin process, fork () returned %d", pid);
+		return -1;
+	}
+
+	pp->pid = pid;
+	ELEKTRA_LOG_DEBUG ("The pluginprocess is set using the pipes %s and %s with the pid %d", pp->commandPipe, pp->resultPipe, pid);
+	return 0;
+}
+
+static char * getPipename (Plugin * handle, char * suffix)
+{
+	int pipenameSize;
+	char * pipename;
+	pipenameSize = snprintf (NULL, 0, "/tmp/libelektra-process-%p-%s", (void *)handle, suffix);
```

Comment:
To use the handle as only distinction is maybe a good idea on systems with Address space layout randomization but we have no guarantees for that. So we should encode some additional randomness in this path. Did you consider to use mkdtemp? (see tests/cframework/tests.c 95). Alternatively you can take a look at elektraGenTempFilename (which could be called by libinvoke)